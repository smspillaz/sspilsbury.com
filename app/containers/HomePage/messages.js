/*
 * HomePage Messages
 *
 * This contains all the text for the HomePage component.
 */
import { defineMessages } from 'react-intl';

export const scope = 'boilerplate.containers.HomePage';

export default defineMessages({
  make: {
    id: 'conatiners.HomePage.header.make',
    defaultMessage: 'Make',
  },
  imagination: {
    id: 'conatiners.HomePage.header.imagination',
    defaultMessage: 'Imagination',
  },
  reality: {
    id: 'conatiners.HomePage.header.reality',
    defaultMessage: 'Reality',
  },
  posts: {
    id: 'containers.HomePage.posts.title',
    defaultMessage: 'Posts',
  },
  projects: {
    id: 'containers.HomePage.projects.title',
    defaultMessage: 'Projects',
  },
  talks: {
    id: 'containers.HomePage.talks.title',
    defaultMessage: 'Talks & Papers',
  },
});
