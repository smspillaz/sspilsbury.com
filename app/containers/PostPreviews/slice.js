import { createSlice } from '@reduxjs/toolkit';

export const initialState = {
  posts: [],
  loading: false,
  error: null,
};

const postPreviewsSlice = createSlice({
  name: 'postPreviews',
  initialState,
  reducers: {
    loadPosts(draft) {
      draft.loading = true;
      draft.error = null;
      draft.posts = [];
    },

    loadPostsSuccess(draft, action) {
      const posts = action.payload;
      draft.loading = false;
      draft.posts = posts;
    },

    loadPostsError(draft, action) {
      draft.error = {
        msg: action.payload.message,
      };
      draft.loading = false;
    },
  },
});

export const {
  loadPosts,
  loadPostsSuccess,
  loadPostsError,
} = postPreviewsSlice.actions;
export const { reducer } = postPreviewsSlice;
