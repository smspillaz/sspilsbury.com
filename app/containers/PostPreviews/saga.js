/**
 * Gets the repositories of the user from Github
 */

import { call, put, takeLatest } from 'redux-saga/effects';
import request from 'utils/request';

import { loadPosts, loadPostsError, loadPostsSuccess } from './slice';

/**
 * Blog posts request/response handler
 */
export function* getPosts({ payload: { wordpress } }) {
  const requestURL = `${process.env.API_HOST ||
    ''}/api/post-previews?wordpressDomain=${wordpress.domain}&wordpressCount=${
    wordpress.count
  }`;

  try {
    // Call our request helper (see 'utils/request')
    const { payload } = yield call(request, requestURL);
    yield put(loadPostsSuccess(payload));
  } catch (err) {
    console.error(err);
    yield put(loadPostsError(err));
  }
}

/**
 * Root saga manages watcher lifecycle
 */
export default function* postsData() {
  // Watches for loadPosts actions and calls getPosts when one comes in.
  // By using `takeLatest` only the result of the latest API call is applied.
  // It returns task descriptor (just like fork) so we can continue execution
  // It will be cancelled automatically on component unmount
  yield takeLatest(loadPosts.type, getPosts);
}
