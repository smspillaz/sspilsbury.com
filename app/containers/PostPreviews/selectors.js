/**
 * The global state selectors
 */

import { createSelector } from 'reselect';
import { initialState } from './slice';

const selectPosts = state => state.posts || initialState;

const selectRouter = state => state.router;

const makeSelectPostsLoading = () =>
  createSelector(selectPosts, postsState => postsState.loading);

const makeSelectPostsError = () =>
  createSelector(selectPosts, postsState => postsState.error);

const makeSelectPosts = () =>
  createSelector(selectPosts, postsState => postsState.posts);

const makeSelectLocation = () =>
  createSelector(selectRouter, routerState => routerState.location);

export {
  selectPosts,
  makeSelectPostsLoading,
  makeSelectPostsError,
  makeSelectPosts,
  makeSelectLocation,
};
