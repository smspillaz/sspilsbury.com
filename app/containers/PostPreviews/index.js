import React, { useEffect } from 'react';
import PropTypes from 'prop-types';

import { connect } from 'react-redux';
import { useInjectReducer, useInjectSaga } from 'redux-injectors';
import { createStructuredSelector } from 'reselect';

import { reducer, loadPosts } from './slice';
import {
  makeSelectPostsLoading,
  makeSelectPostsError,
  makeSelectPosts,
} from './selectors';
import saga from './saga';

export const PostPreviews = React.createContext();

const PostPreviewsFetcher = ({
  children,
  loadPostsAction,
  medium,
  wordpress,
  loading,
  error,
  posts,
}) => {
  useInjectReducer({ key: 'posts', reducer });
  useInjectSaga({ key: 'posts', saga });
  useEffect(() => {
    loadPostsAction({
      medium,
      wordpress,
    });
  }, [medium, wordpress]);

  return (
    <PostPreviews.Provider
      value={{
        loading,
        error,
        posts,
      }}
    >
      {children}
    </PostPreviews.Provider>
  );
};

PostPreviewsFetcher.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.node,
    PropTypes.arrayOf(PropTypes.node),
  ]),
  loadPostsAction: PropTypes.func.isRequired,
  medium: PropTypes.shape({
    username: PropTypes.string.isRequired,
    count: PropTypes.number.isRequired,
  }),
  wordpress: PropTypes.shape({
    domain: PropTypes.string.isRequired,
    count: PropTypes.number.isRequired,
  }),
  loading: PropTypes.bool.isRequired,
  error: PropTypes.string,
  posts: PropTypes.array.isRequired,
};

const mapStateToProps = createStructuredSelector({
  loading: makeSelectPostsLoading(),
  error: makeSelectPostsError(),
  posts: makeSelectPosts(),
});
const mapDispatchToProps = dispatch => ({
  loadPostsAction: ({ wordpress, medium }) =>
    dispatch(loadPosts({ wordpress, medium })),
});

const withConnect = connect(mapStateToProps, mapDispatchToProps);
const ConnectedPostPreviewsFetcher = withConnect(PostPreviewsFetcher);
ConnectedPostPreviewsFetcher.displayName = 'ConnectedPostPreviewsFetcher';

export default ConnectedPostPreviewsFetcher;
