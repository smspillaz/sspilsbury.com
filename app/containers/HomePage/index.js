/*
 * HomePage
 *
 * This is the first thing users see of our App, at the '/' route
 *
 */

import React from 'react';
import { Helmet } from 'react-helmet';
import { FormattedMessage } from 'react-intl';
import PropTypes from 'prop-types';

import { Heading, Text } from 'rebass/styled-components';
import { Hero, Section } from 'react-landing-page';

import { ArticleContent } from 'components/ArticleContent';
import { BackgroundAnimation } from 'components/BackgroundAnimation';
import LoadingIndicator from 'components/LoadingIndicator';
import { PostPreview } from 'components/PostPreview';
import { ProjectPreview } from 'components/ProjectPreview';
import { ProjectPreviewImageCarousel } from 'components/ProjectPreviewImageCarousel';
import { LayoutTypes } from 'components/ResponsiveOrderedLayoutChild';
import { VisibilityTrackedScrollDownIndicator } from 'components/VisibilityTrackedScrollDownIndicator';
import { PostPreviews } from 'containers/PostPreviews';
import messages from './messages';
import { projects } from '../../content/projects';

export default function HomePage({ animating = true }) {
  return (
    <article>
      <Helmet>
        <title>Sam Spilsbury</title>
        <meta name="description" content="Sam Spilsbury's Portfolio" />
      </Helmet>
      <Hero color="black" bgOpacity={0.0}>
        <BackgroundAnimation
          style={{
            margin: '0 0 0 0',
            zIndex: 0,
            width: '100%',
            height: '100%',
            position: 'absolute',
          }}
          animating={animating}
        />
        <ArticleContent style={{ textAlign: 'center' }}>
          <Heading fontSize={[5, 6, 7]}>
            <FormattedMessage {...messages.make} />
          </Heading>
          <Heading fontSize={[5, 6, 7]}>
            <FormattedMessage {...messages.imagination} />
          </Heading>
          <Heading fontSize={[5, 6, 7]}>
            <FormattedMessage {...messages.reality} />
          </Heading>
          <VisibilityTrackedScrollDownIndicator />
        </ArticleContent>
      </Hero>
      <ArticleContent>
        <Section width={1} heading={messages.posts.defaultMessage}>
          <PostPreviews.Consumer>
            {({ loading, error, posts }) => (
              <div>
                {loading && <LoadingIndicator />}
                {error && <Text>{error.msg}</Text>}
                {!error &&
                  posts
                    .slice(0, 5)
                    .map(post => (
                      <PostPreview
                        title={post.title}
                        subtitle={post.subtitle}
                        date={post.date}
                        likes={post.likes}
                        words={post.words}
                        url={post.url}
                        key={post.url}
                      />
                    ))}
              </div>
            )}
          </PostPreviews.Consumer>
        </Section>
        <Section width={1} heading={messages.projects.defaultMessage}>
          {projects
            .filter(({ featured }) => featured)
            .map((project, i) => (
              <ProjectPreview
                header={project.title}
                subtitle={project.subtitle}
                client={project.client}
                description={project.short}
                layout={i % 2 === 0 ? LayoutTypes.LEFT : LayoutTypes.RIGHT}
                url={project.url}
                key={project.key}
                renderContractedPreview={() => (
                  <ProjectPreviewImageCarousel
                    images={project.images}
                    imageStyleProps={project.imageStyleProps || {}}
                  />
                )}
              />
            ))}
        </Section>
      </ArticleContent>
    </article>
  );
}

HomePage.propTypes = {
  animating: PropTypes.bool,
};
