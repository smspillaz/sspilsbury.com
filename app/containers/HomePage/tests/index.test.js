import React from 'react';
import { render } from '@testing-library/react';
import { IntlProvider } from 'react-intl';
import { ThemeProvider } from 'styled-components';

import preset from '@rebass/preset';

import HomePage from '../index';
import { PostPreviews } from '../../PostPreviews';

describe('<HomePage />', () => {
  it('should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      <IntlProvider locale="en">
        <ThemeProvider theme={preset}>
          <PostPreviews.Provider
            value={{
              loading: false,
              error: null,
              posts: [],
            }}
          >
            <HomePage animating={false} />
          </PostPreviews.Provider>
        </ThemeProvider>
      </IntlProvider>,
    );
    expect(firstChild).toMatchSnapshot();
  });
});
