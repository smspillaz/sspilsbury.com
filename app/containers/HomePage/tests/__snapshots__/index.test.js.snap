// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomePage /> should render and match the snapshot 1`] = `
<article>
  <div
    class="sc-cJSrbW dLwzxs sc-ksYbfQ hQmkLk"
  >
    <div
      class="sc-cJSrbW gDgscr sc-hmzhuo heFgru"
      style="opacity: 1; height: 100%;"
    />
    <div
      class="sc-htpNat kjEEUt sc-bxivhb kPjOIj"
      style="min-height: 100vh;"
    >
      <div
        style="margin: 0px 0px 0px 0px; z-index: 0; width: 100%; height: 100%; position: absolute;"
      >
        <div
          class="AnimatedCanvas__SizeTrackingContainer-sc-1fa2asd-0 hPyter"
          style="margin: 0px 0px 0px 0px; z-index: 0; width: 100%; height: 100%; position: absolute;"
        >
          <canvas
            height="0"
            style="margin: 0px 0px 0px 0px; z-index: 0; width: 100%; height: 100%; position: absolute;"
            width="0"
          />
        </div>
      </div>
      <div
        class="ArticleContent-sc-8xf5qw-0 bbqeai"
        style="text-align: center;"
      >
        <h2
          class="sc-bdVaJa jbjLLb"
          font-size="5,6,7"
        >
          <span>
            Make
          </span>
        </h2>
        <h2
          class="sc-bdVaJa jbjLLb"
          font-size="5,6,7"
        >
          <span>
            Imagination
          </span>
        </h2>
        <h2
          class="sc-bdVaJa jbjLLb"
          font-size="5,6,7"
        >
          <span>
            Reality
          </span>
        </h2>
        <div>
          <div />
        </div>
      </div>
    </div>
  </div>
  <div
    class="ArticleContent-sc-8xf5qw-0 bbqeai"
  >
    <div
      class="sc-htpNat bLaVye sc-bxivhb jlzawt"
    >
      <h2
        class="sc-fjdhpX cXJGRi"
      >
        Posts
      </h2>
      <h3
        class="sc-fjdhpX gdrLVw sc-jzJRlG enxKql"
        style="opacity: 0.5;"
      />
      <div />
    </div>
    <div
      class="sc-htpNat bLaVye sc-bxivhb jlzawt"
    >
      <h2
        class="sc-fjdhpX cXJGRi"
      >
        Projects
      </h2>
      <h3
        class="sc-fjdhpX gdrLVw sc-jzJRlG enxKql"
        style="opacity: 0.5;"
      />
      <div
        class="sc-bdVaJa sc-bwzfXH ProjectPreview__ResponsiveFlex-sc-1cs94to-0 kecwqy"
        direction="row"
        width="1"
      >
        <div
          class="ResponsiveOrderedLayoutChild-sc-9hvmti-0 bWWyfP"
          style="align-self: flex-begin;"
          width="1,1,1,0.4"
        >
          <div
            class="ImageRelativeParent-sc-8zyta2-0 knMrFP"
          >
            <div>
              <div
                class="ImageAbsoluteParent-fqa90c-0 lmjMCH"
                style="background-image: url(https://s3.amazonaws.com/sspilsbury-com-images/project-intuimath-intersections.gif); background-size: cover; background-position: center center; border-radius: 8px; transform: scale(0.9); opacity: 1; transition: all 1.5s ease-in-out;"
              />
              <div
                class="ImageAbsoluteParent-fqa90c-0 lmjMCH"
                style="background-image: url(https://s3.amazonaws.com/sspilsbury-com-images/project-intuimath-surface-area.gif); background-size: cover; background-position: center center; border-radius: 8px; opacity: 0; filter: blur(16px); transition: all 1.5s ease-in-out;"
              />
            </div>
          </div>
        </div>
        <div
          class="ResponsiveOrderedLayoutChild-sc-9hvmti-0 iycRdG"
          width="1,1,1,0.5"
        >
          <a
            class="Link-sc-1qsuhxf-0 jAEqpf"
            color="black"
            href="https://intuitive-math.club"
          >
            <h2
              class="sc-bdVaJa jrtyUD"
            >
              Intuitive Math
            </h2>
          </a>
          <h2
            class="sc-bdVaJa enmavF"
            color="grey"
            font-size="1,2,3"
          >
            Animated math primer for graphics
          </h2>
          <a
            class="Link-sc-1qsuhxf-0 XoWQB"
            color="grey"
            href="https://intuitive-math.club"
          >
            <div
              class="sc-bdVaJa gtVQmG"
            >
              intuitive-math.club
            </div>
          </a>
          <div
            class="MarkdownChildrenStyles-sc-1xr5379-0 eqSurb"
          >
            <div
              class="sc-bdVaJa gtVQmG"
            >
              Intuitive Math is a hosted primer for graphics math, primarily linear algebra and calculus.
It takes its inspiration from 3blue1brown, where it combines worked examples of problems along with their visual intuition at each step in the process.
            </div>
          </div>
        </div>
      </div>
      <div
        class="sc-bdVaJa sc-bwzfXH ProjectPreview__ResponsiveFlex-sc-1cs94to-0 kecwqy"
        direction="row"
        width="1"
      >
        <div
          class="ResponsiveOrderedLayoutChild-sc-9hvmti-0 bWWyfP"
          style="align-self: flex-end;"
          width="1,1,1,0.4"
        >
          <div
            class="ImageRelativeParent-sc-8zyta2-0 knMrFP"
          >
            <div>
              <div
                class="ImageAbsoluteParent-fqa90c-0 lmjMCH"
                style="background-image: url(https://s3.amazonaws.com/sspilsbury-com-images/project-jurimetrics-cases-over-time.png); background-size: cover; background-position: center center; border-radius: 8px; transform: scale(0.9); opacity: 1; transition: all 1.5s ease-in-out;"
              />
              <div
                class="ImageAbsoluteParent-fqa90c-0 lmjMCH"
                style="background-image: url(https://s3.amazonaws.com/sspilsbury-com-images/project-jurimetrics-federal-court-cases.png); background-size: cover; background-position: center center; border-radius: 8px; opacity: 0; filter: blur(16px); transition: all 1.5s ease-in-out;"
              />
              <div
                class="ImageAbsoluteParent-fqa90c-0 lmjMCH"
                style="background-image: url(https://s3.amazonaws.com/sspilsbury-com-images/project-jurimetrics-feed.png); background-size: cover; background-position: center center; border-radius: 8px; opacity: 0; filter: blur(16px); transition: all 1.5s ease-in-out;"
              />
            </div>
          </div>
        </div>
        <div
          class="ResponsiveOrderedLayoutChild-sc-9hvmti-0 fLLNBd"
          width="1,1,1,0.5"
        >
          <a
            class="Link-sc-1qsuhxf-0 jAEqpf"
            color="black"
            href="https://demo.jurimetrics.com.au"
          >
            <h2
              class="sc-bdVaJa jrtyUD"
            >
              Jurimetrics
            </h2>
          </a>
          <h2
            class="sc-bdVaJa enmavF"
            color="grey"
            font-size="1,2,3"
          >
            Law and Government in Australia
          </h2>
          <a
            class="Link-sc-1qsuhxf-0 XoWQB"
            color="grey"
            href="https://jurimetrics.com.au"
          >
            <div
              class="sc-bdVaJa gtVQmG"
            >
              Jurimetrics Pty Ltd
            </div>
          </a>
          <div
            class="MarkdownChildrenStyles-sc-1xr5379-0 eqSurb"
          >
            <div
              class="sc-bdVaJa gtVQmG"
            >
              Jurimetrics is a platform and database compiled from hundreds of thousands of publically available and licensed case reports, prospectii and government registers. See the public legal and regulatory record for almost any company in Australia
            </div>
          </div>
        </div>
      </div>
      <div
        class="sc-bdVaJa sc-bwzfXH ProjectPreview__ResponsiveFlex-sc-1cs94to-0 kecwqy"
        direction="row"
        width="1"
      >
        <div
          class="ResponsiveOrderedLayoutChild-sc-9hvmti-0 bWWyfP"
          style="align-self: flex-begin;"
          width="1,1,1,0.4"
        >
          <div
            class="ImageRelativeParent-sc-8zyta2-0 knMrFP"
          >
            <div>
              <div
                class="ImageAbsoluteParent-fqa90c-0 lmjMCH"
                style="background-image: url(https://s3.amazonaws.com/sspilsbury-com-images/project-endless-libanimation-window-animations.gif); background-size: cover; background-position: center center; border-radius: 8px; transform: scale(0.9); opacity: 1; transition: all 1.5s ease-in-out;"
              />
              <div
                class="ImageAbsoluteParent-fqa90c-0 lmjMCH"
                style="background-image: url(https://s3.amazonaws.com/sspilsbury-com-images/project-endless-libanimation-wobbly-windows.gif); background-size: cover; background-position: center center; border-radius: 8px; opacity: 0; filter: blur(16px); transition: all 1.5s ease-in-out;"
              />
            </div>
          </div>
        </div>
        <div
          class="ResponsiveOrderedLayoutChild-sc-9hvmti-0 iycRdG"
          width="1,1,1,0.5"
        >
          <a
            class="Link-sc-1qsuhxf-0 jAEqpf"
            color="black"
            href="https://github.com/smspillaz/libanimation"
          >
            <h2
              class="sc-bdVaJa jrtyUD"
            >
              libanimation
            </h2>
          </a>
          <h2
            class="sc-bdVaJa enmavF"
            color="grey"
            font-size="1,2,3"
          >
            Math for window animations
          </h2>
          <a
            class="Link-sc-1qsuhxf-0 XoWQB"
            color="grey"
            href="https://endlessos.com"
          >
            <div
              class="sc-bdVaJa gtVQmG"
            >
              Endless
            </div>
          </a>
          <div
            class="MarkdownChildrenStyles-sc-1xr5379-0 eqSurb"
          >
            <div
              class="sc-bdVaJa gtVQmG"
            >
              libanimation is a math library that can be used by any compositor to provide nice looking window animations. It provides the math for wobbly windows, a magic lamp effect and some simple affine transformations.
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</article>
`;
